### PR TITLE
initial commit for zoom api key cycling

### PR DESCRIPTION
**Pull request by Maosef**
Address ticket https://trello.com/c/LL3aBYci/361-schedule-implement-zoom-account-cycling-mechanism-to-circumvent-api-rate-limits. Modify `createMeetingHelper` to access zoom API keys from a DynamoDB database, and cycle though them when encountering an error.

### CODE REVIEW

File: backend/schedule/config/prod.json [hunk 1]
```diff
@@ -7,5 +7,6 @@
     "ZOOM_API_EMAIL_ACCOUNT": "hugoburbelo@gmail.com",
     "ZOOM_LINK_TABLE": "platform-prod-meetings",
     "ZOOM_API_KEY_SECRET_NAME": "ZOOM_TESTING_API_KEY",
-    "PLATFORM_BASE_URL": "http://platform.gotechnica.org"
+    "PLATFORM_BASE_URL": "http://platform.gotechnica.org",
+    "ZOOM_API_KEYS_TABLE": "platform-dev-zoom-api-keys"
```
Comment (burbelohugo) [inline backend/schedule/config/prod.json:11]: I think these need to be a different stage than dev, right? `platform-prod-zoom-api-keys` and so on

File: backend/schedule/handler.js [hunk 2]
```diff
@@ -498,7 +512,7 @@ module.exports.get_events_from_user_list = withSentry(async (event) => {
 
 module.exports.create_zoom_meeting = withSentry(async (event) => {
   const body = JSON.parse(event.body);
-
+  // const body = event.body;
```
Comment (burbelohugo) [inline backend/schedule/handler.js:515]: Let's remove this comment, please

File: backend/infrastructure/add_zoom_keys.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+const TABLE_LIST = [
+  { name: 'zoom-api-keys', keyName: 'key' },
+
+];
+
+const STAGE_LIST = ['prod', 'stage', 'dev', 'test'];
+const KEY_LIST = [
+    {
+        "key": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOm51bGwsImlzcyI6IjFpWDhRUmZzVDVtLUpqYW5DSkZITGciLCJleHAiOjE2Mjc5MTUwMjAsImlhdCI6MTU5NTc3NDI1Mn0._pS6-Jn75VDxRZlO3iKcwLsIV48ppXTNXN_CAM5dPLE",
```
Comment (burbelohugo) [inline backend/infrastructure/add_zoom_keys.js:18]: Please remove the raw key and email and add text placeholders, we don't want to be committing them publicly to Github

File: backend/infrastructure/add_zoom_keys.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        "email": "hugoburbelo@gmail.com"
+    },
+    {
+        "key": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOm51bGwsImlzcyI6IlQydVRsMHd2U01pa2JheUFhbzIxbGciLCJleHAiOjE2MzAyOTU3MDAsImlhdCI6MTU5ODc1NDMzMX0.iYFHbhoLSdBl7O3FHj6_cWwNZ_IflPLP0uZv-iYwabY",
+        "email": "amao1@terpmail.umd.edu"
+    }
+]
+
+const createTable = (tableName, keyName, stage) => {
```
Comment (burbelohugo) [inline backend/infrastructure/add_zoom_keys.js:27]: Is this used? Isn't this script to add keys?

File: backend/infrastructure/add_zoom_keys.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
+const AWS = require('aws-sdk');
```
Comment (burbelohugo) [inline backend/infrastructure/add_zoom_keys.js:1]: Can you add an explanatory comment at the top of the file explaining what this script is used for and how to use/run it, please?

File: backend/infrastructure/add_zoom_keys.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    // Call DynamoDB to add the item to the table
+    await ddb.put(params).promise();
+
+    console.log("success");
+};
+
+// create tables, then populate
+STAGE_LIST.forEach((stage) => {
+  TABLE_LIST.forEach((table) => {
+    // createTable(table.name, table.keyName, stage);
```
Comment (burbelohugo) [inline backend/infrastructure/add_zoom_keys.js:81]: Please remove this comment if it's not used, thanks

File: backend/infrastructure/cycle_zoom.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
+var rp = require('request-promise');
```
Comment (burbelohugo) [inline backend/infrastructure/cycle_zoom.js:1]: Sorry, I'm not clear on what this file is used for or if it should be included. If this functionality already exists as code in a handler, I'm not sure what the benefit of keeping it is - can we remove it unless there's a good reason for preserving it?

File: backend/infrastructure/package.json [hunk 8]
```diff
@@ -10,7 +10,8 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "aws-sdk": "^2.714.0"
+    "aws-sdk": "^2.714.0",
+    "request-promise": "^4.2.6"
```
Comment (burbelohugo) [inline backend/infrastructure/package.json:14]: If we're not going to use the above script, let's get rid of this

File: backend/infrastructure/add_zoom_keys.js (hunk 4 above)
Comment (Maosef) [reply backend/infrastructure/add_zoom_keys.js:27]: Correct, the script is used for automatically adding keys. Can be used when we get more keys.

File: backend/infrastructure/cycle_zoom.js (hunk 7 above)
Comment (Maosef) [reply backend/infrastructure/cycle_zoom.js:1]: this script enables quickly testing the API keys until the limit is reached. But yes it's more for personal testing.

File: backend/infrastructure/add_zoom_keys.js (hunk 4 above)
Comment (burbelohugo) [reply backend/infrastructure/add_zoom_keys.js:27]: Sorry, I was talking about the createTable function on line 7 and whether it's used

Comment (Maosef) [reply backend/infrastructure/add_zoom_keys.js:26]: yes it is used if the tables get wiped. Similar to `create-tables.js`

**Comment by burbelohugo** [comment]
Amazing! 💯

**Pull request merged**